image/descriptor.go

```diff
@@ -50,17 +51,26 @@ func (d *descriptor) hash() string {
 
 func listReferences(w walker) (map[string]*descriptor, error) {
 	refs := make(map[string]*descriptor)
+	var index v1.ImageIndex
 
 	if err := w.walk(func(path string, info os.FileInfo, r io.Reader) error {
-		if info.IsDir() || !strings.HasPrefix(path, "refs") {
+		if info.IsDir() || filepath.Clean(path) != "index.json" {
 			return nil
 		}
 
-		var d descriptor
-		if err := json.NewDecoder(r).Decode(&d); err != nil {
+		if err := json.NewDecoder(r).Decode(&index); err != nil {
 			return err
 		}
-		refs[info.Name()] = &d
+
+		var d descriptor
+		for i := 0; i < len(index.Manifests); i++ {
+			if index.Manifests[i].Descriptor.Annotations["org.opencontainers.ref.name"] != "" {
+				d.MediaType = index.Manifests[i].Descriptor.MediaType
+				d.Digest = string(index.Manifests[i].Descriptor.Digest)
+				d.Size = index.Manifests[i].Descriptor.Size
+				refs[index.Manifests[i].Descriptor.Annotations["org.opencontainers.ref.name"]] = &d
```

Comment:
@stevvooe Sorry to disturb, I have solved other problems, only the problem can not be resolved. 
> descriptor.go:64: cannot use index.Manifests[i].Descriptor (type v1.Descriptor) as type descriptor in assignment

Or is there a problem with my method. Can we discuss this question? If can't solve the problem I think I still use the method in https://github.com/opencontainers/image-tools/pull/144/commits/ea52a5986f7dcd7af63519784b60debf85ebddae.  （ I think this method is feasible.）
I hope to also try to promote the development of image-tool, but sometimes some technical problems need your help.Thanks.